### PR TITLE
JBMAR-193 Filter Incoming Marshalling Data

### PR DESCRIPTION
**Pull request by soul2zimate**
https://issues.jboss.org/browse/JBMAR-193 Filter Incoming Marshalling Data

This one tries to complete previous work in https://github.com/jboss-remoting/jboss-marshalling/pull/86

### CODE REVIEW

**Comment by xstefank** [comment]
@dmlloyd this has a downstream dependency, can you take a look when you will be available, please?

File: api/src/main/java/org/jboss/marshalling/InputFilter.java [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Filter that implementations of {@link Unmarshaller} can apply after resolving the class to decide if deserialization has to stop.
+ */
+@FunctionalInterface
+public interface InputFilter {
```
Comment (dmlloyd) [inline api/src/main/java/org/jboss/marshalling/InputFilter.java]: The formatting in this class seems inconsistent.  Please reformat with a uniform four-space indent.

File: api/src/main/java/org/jboss/marshalling/InputFilter.java [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+    long depth();
+
+    long references();
+
+    long streamBytes();
+
+  }
+
+  final static class Global implements InputFilter {
```
Comment (dmlloyd) [inline api/src/main/java/org/jboss/marshalling/InputFilter.java]: Classes nested within interfaces are always `static`.

File: api/src/main/java/org/jboss/marshalling/InputFilter.java [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+              // } else {
+              // filters.add(c -> moduleName.equals(c.getModule().getName()) ? patternFilter.apply(c) : Status.UNDECIDED);
+              // }
+              filters.add(patternFilter);
+          }
+      }
+
+      /**
+       * Returns if this filter has any checks.
+       * @return {@code true} if the filter has any checks, {@code false} otherwise
```
Comment (dmlloyd) [inline api/src/main/java/org/jboss/marshalling/InputFilter.java]: I think this description is backwards.

**Comment by soul2zimate** [comment]
Close, this need to rework

**Pull request closed without merging**